CoEpiTests/FFISanityTests.swift

```diff
@@ -37,57 +37,89 @@ class FFISanityTests: XCTestCase {
         }
         let returned_struct = pass_and_return_struct(pointer)
         XCTAssertEqual(returned_struct.my_int, myStruct.my_int)
-
+        
         let unmanagedString: Unmanaged<CFString> = returned_struct.my_str
         let cfStr: CFString = unmanagedString.takeRetainedValue()
         let returnedStr = cfStr as String
-
+        
         XCTAssertEqual(returnedStr, str)
         XCTAssertEqual(returned_struct.my_nested.my_u8, myStruct.my_nested.my_u8)
     }
-
+    
     func testCallback() {
         let ret = call_callback { (myInt, myBool, myCFString) in
             // TODO why CFString? here while in the struct it's Unmanaged<CFString>
             let cfStr: CFString = myCFString!
             let str = cfStr as String
-
+            
             XCTAssertEqual(myInt, 123)
             XCTAssertEqual(myBool, false)
             XCTAssertEqual(str, "hi!")
         }
         XCTAssertEqual(ret, 1)
     }
-
-
+    
+    
     func testRegisterCallback() {
         expectationWorkaroundGlobal = self.expectation(description: #function)
-
+        
         let ret = register_callback { (myInt, myBool, myCFString) in
             // TODO why CFString? here while in the struct it's Unmanaged<CFString>
             let cfStr: CFString = myCFString!
             let str = cfStr as String
-
+            
             // The first 2 parameters are hardcoded in Rust
             XCTAssertEqual(myInt, 1)
             XCTAssertEqual(myBool, true)
-
+            
             // This is what we send here
             XCTAssertEqual(str, "hi from iOS")
-
-            // NOTE: MAYOR WORKAROUND: for:
+            
+            // NOTE: MAJOR WORKAROUND: for:
             // "A C function pointer cannot be formed from a closure that captures context"
             // It only accepts global variables.
             // There's a better workaround IIRC, but more complex. For now letting it like this.
             // TODO better workaround.
             expectationWorkaroundGlobal?.fulfill()
         }
         XCTAssertEqual(ret, 1)
-
+        
         let triggerRet = trigger_callback("hi from iOS")
         XCTAssertEqual(triggerRet, 1)
-
+        
         waitForExpectations(timeout: 5, handler: nil)
     }
+    
+    func testRegisterCallbackForLogging() {
+        let ret = register_callback { (myInt, myBool, myCFString) in
+            let cfStr: CFString = myCFString!
+            let str = cfStr as String
+            
+            // The first 2 parameters are hardcoded in Rust
+            XCTAssertEqual(myInt, 1)
+            XCTAssertEqual(myBool, true)
```

Comment:
This shouldn't be part of the logger functionality/tests.

Reply:
Method deleted (it was testing string logging)